Dispatcher/UI/Search/SearchViewModel.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  SearchScreenViewModelController.swift
+//  Dispatcher
+//
+//  Created by Ofek Cohen on 02/05/2023.
+//
+
+import Foundation
+
+class SearchViewModel {
+    var articlesToDisplay: [Article] = []
+    var latestSearches: [String] = []
+    var totalResultsPages = 1
+    var isPaginating = false
+    
+    func getArticlesFromAPIBySearch(_ searchKeyWords: String, pageNumber: Int, completionHandler: @escaping (_ errorMsg: String?) -> Void) {
+        if (isPaginating == false) {
+            isPaginating = true
+            AppConstants.articlesRepository.getArticlesByKeywords(searchKeyWords, pageNumber: pageNumber, completionHandler: { articles, totalPages, errorMsg in
+                if (errorMsg != nil) {
+                    completionHandler(errorMsg)
+                } else {
+                    self.articlesToDisplay = articles
+                    self.totalResultsPages = totalPages
+                    self.isPaginating = false
+                    completionHandler(nil)
+                }
+            })
+        }
+    }
+    
+    func articlesAmount() -> Int {
+        return articlesToDisplay.count
+    }
+    
+    func currentArticle(index: Int) -> Article {
+        return articlesToDisplay[index]
+    }
+    
+    func fetchLatestSearchs() {
+        latestSearches = (AppConstants.defaults.array(forKey: AppConstants.latestSearchesDefualtsKey) as? [String]) ?? []
```

Comment:
We want a search repository that will communicate with our UserDefaultsManager to comply with the MVVM+repository architecture that we have in our project